go/common/enclave.go

```diff
@@ -130,3 +132,22 @@ type ProducedSecretResponse struct {
 	RequesterID gethcommon.Address
 	HostAddress string
 }
+
+// BlockRejectError is used as a standard format for error response from enclave for block submission errors
+// The L1 Head hash tells the host what the enclave knows as the canonical chain head, so it can feed it the appropriate block.
+type BlockRejectError struct {
+	L1Head  gethcommon.Hash
+	Wrapped error
```

Comment:
Since you're forced to have the `Unwrap` method, maybe make this field private and force the use of that method?

Reply:
Yeah kinda agree but would need a method to populate it as well so will probably just leave it. I hope this stuff doesn't live too long tbh.